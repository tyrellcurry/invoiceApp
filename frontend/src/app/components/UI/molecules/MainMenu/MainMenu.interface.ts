export interface IMainMenuProps {
  darkmode: string | undefined;
  darkmodeToggle: (event: React.MouseEvent<HTMLElement, MouseEvent>) => void;
  profileImage?: string;
  profileImageAlt: string;
  darkmodeBtnAria?: string;
  lightmodeBtnAria?: string;
}
