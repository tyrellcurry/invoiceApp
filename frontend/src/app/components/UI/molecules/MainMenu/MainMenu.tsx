import React, { JSX } from 'react';
import { IMainMenuProps } from '@/app/components/UI/molecules/MainMenu/MainMenu.interface';
import Button from '@/app/components/UI/atoms/Button/Button';
import Image from 'next/image';
const MainMenu = (props: IMainMenuProps): JSX.Element => {
  const { darkmode, darkmodeToggle, profileImage, profileImageAlt } = props;
  return (
    <nav className="bg-gray-09 flex justify-between items-stretch lg:flex-col lg:w-fit lg:rounded-r-[20px] lg:rounded-tr-3xl lg:h-screen lg:fixed lg:left-0">
      <Button
        className="w-[50px] h-[50px] xs:w-[72px] xs:h-[72px] md:w-[80px] md:h-[80px]"
        iconLeft={'menu-logo-btn'}
        iconLeftClassName="w-full h-full"
        variant="custom"
      />
      <div className="flex gap-5 px-6 md:gap-8 md:px-8 lg:flex-col lg:gap-5 lg:px-0 lg:py-6">
        <Button
          className="h-fit self-center"
          iconLeft={darkmode ? 'dm-sun' : 'dm-moon'}
          iconLeftClassName="w-8 h-8 fill-gray-07 hover:fill-gray-05"
          variant="custom"
          onClick={() => darkmodeToggle}
        />
        <hr className="border-none w-0.5 h-full bg-gray-04 self-stretch lg:w-full lg:h-0.5" />
        <Button className="h-fit self-center relative" variant="custom">
          <Image
            className="rounded-full w-full h-auto max-w-8"
            src={!!profileImage ? profileImage : '/assets/profile-default.png'}
            alt={profileImageAlt}
            width={64}
            height={64}
          />
        </Button>
      </div>
    </nav>
  );
};

export default MainMenu;
