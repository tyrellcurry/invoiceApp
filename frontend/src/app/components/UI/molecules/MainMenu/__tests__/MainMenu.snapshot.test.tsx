import React from 'react';
import { render } from '@testing-library/react';
import MainMenu from '@/app/components/UI/molecules/MainMenu/MainMenu';

describe('Main Menu Component', () => {
  it('renders correctly with default props', () => {
    const { container } = render(
      <MainMenu
        darkmode={'dark'}
        darkmodeToggle={() => {}}
        profileImage={'/assets/profile-default.png'}
        profileImageAlt="lorem ipsum"
      />
    );
    expect(container).toMatchSnapshot();
  });
});
