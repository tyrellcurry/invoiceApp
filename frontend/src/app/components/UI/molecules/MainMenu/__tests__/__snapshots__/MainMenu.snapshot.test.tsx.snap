// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Menu Component renders correctly with default props 1`] = `
<div>
  <nav
    class="bg-gray-09 flex justify-between items-stretch lg:flex-col lg:w-fit lg:rounded-r-[20px] lg:rounded-tr-3xl lg:h-screen lg:fixed lg:left-0"
  >
    <a
      class="btn btn--custom w-[50px] h-[50px] xs:w-[72px] xs:h-[72px] md:w-[80px] md:h-[80px]"
    >
      <svg
        class="icon w-full h-full"
      />
    </a>
    <div
      class="flex gap-5 px-6 md:gap-8 md:px-8 lg:flex-col lg:gap-5 lg:px-0 lg:py-6"
    >
      <button
        class="btn btn--custom h-fit self-center"
      >
        <svg
          class="icon w-8 h-8 fill-gray-07 hover:fill-gray-05"
        />
      </button>
      <hr
        class="border-none w-0.5 h-full bg-gray-04 self-stretch lg:w-full lg:h-0.5"
      />
      <button
        class="btn btn--custom h-fit self-center relative"
      >
        <img
          alt="lorem ipsum"
          class="rounded-full w-full h-auto max-w-8"
          data-nimg="1"
          decoding="async"
          height="64"
          loading="lazy"
          src="/_next/image?url=%2Fassets%2Fprofile-default.png&w=128&q=75"
          srcset="/_next/image?url=%2Fassets%2Fprofile-default.png&w=64&q=75 1x, /_next/image?url=%2Fassets%2Fprofile-default.png&w=128&q=75 2x"
          style="color: transparent;"
          width="64"
        />
      </button>
    </div>
  </nav>
</div>
`;
